### ISSUE TITLE
Running test suite leaves a top level folder "a-path-that-will-cause-an-OSError"

### ISSUE THREAD

**Issue by MarkusPiotrowski** [beeware/briefcase#1187]
### Describe the bug

Running the test suite on Windows creates a top level folder named "a-path-that-will-cause-an-OSerror", which is not removed after the tests have finished.
E.g.:
D:/
|_ some other folder
|_ briefcase (the dev folder)
|_ a-path-that-will-cause-an-OSError
|_ more folders

### Steps to reproduce

1. Set up a briefcase development environment on Wndows as described here: https://briefcase.readthedocs.io/en/latest/how-to/contribute-code.html
2. Run `tox -e py`

### Expected behavior

All changes outside the dev folder and the /user/.../AppData/ folders should be reverted

### Environment

- Operating System: Windows 10 Home 21H2
- Python version: 3.8.8
- Software versions:
  - Briefcase: 0.3.14.dev151+g27996ba4

### Logs

```

```

**Comment by mhsmith** [comment]
I've also noticed this.

**Comment by rmartin16** [comment]
Yeah. I've been meaning to fix this; thanks for documenting it. I may put together something today.....there's some other small testing issues I've noticed recently as well.
